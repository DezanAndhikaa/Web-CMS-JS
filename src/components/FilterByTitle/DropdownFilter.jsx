

import React from 'react';
import { ExpandMore } from '@material-ui/icons';
import './DropdownFilter.scss';

class DropdownFilter extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      displayMenu: false,
    };
  }

  showDropdownMenu = (event) => {
    event.preventDefault();
    this.setState({ displayMenu: true }, () => {
      document.addEventListener('click', this.hideDropdownMenu);
    });
  }

  hideDropdownMenu = () => {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener('click', this.hideDropdownMenu);
    });
  }

  selectItem = (item,Tab) => {
    this.props.onSelectAction(this.props.onSelectActionType, item);
    console.log("nilai si Tab ", Tab)
    this.props.indexFilter("INDEX FILTER",Tab);
  }

  handleJobType = (item) => {
    let newItem = item
    if(item === 'INS'){
      newItem = 'Periodic Inspection'
    }
    return newItem
  }

  renderDropdown() {
    return (
      <div className="dropdown-filter-button" onClick={this.showDropdownMenu}>
        <div className="dropdown-selected-items">
          {this.props.selected}
          <div className="expand-icon-containers"><ExpandMore className="expand-icons" /></div>
        </div>
      </div>
    );
  }

  renderDropdownList() {
    return (
      <ul className="list-items-filter">
        {this.props.data
          && this.props.data.map((item, index) => (
            item && <div className="list-item-filter" onClick={() => this.selectItem(item,this.props.indexTab)} key={index}>{this.handleJobType(item)}</div>
          ))}
      </ul>
    );
  }

  render() {
    if (this.props.displayMode === 'mobile') {
      return (
        <div>
          <div className="dropdown-filter">
            {this.renderDropdown()}
          </div>
          {
            this.state.displayMenu && this.renderDropdownList()
          }
        </div>
      );
    }

    return (
      <div className="dropdown-filter">
        {this.renderDropdown()}
        {
          this.state.displayMenu && this.renderDropdownList()
        }
      </div>
    );
  }
}

export default DropdownFilter;