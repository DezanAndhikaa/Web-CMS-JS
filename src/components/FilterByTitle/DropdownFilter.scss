@import '../../themes/variables';
@import '../../themes/functions';

.dropdown-filter {
    position: relative;
    display: inline-block;
    background-color: #ffffff;
    width: vw(203.22);
    height: vw(38.07);
    margin-top: vw(41.93); //51.93
    border-radius:vw(10);
    box-shadow: 0px 0.2vw 0.6vw rgba(0, 0, 0, 0.161);
    .dropdown-filter-button {
        border-radius: vw(5);
        display: flex;
        align-items: center;
        height: vw(38.07);
        .dropdown-selected-items{            
            margin: 0 0 0 vw(19);
            display: flex;
            flex-direction: row;
            white-space: nowrap;
            align-items: center;
            font-family: 'Lato-Medium';
            font-size: vw(14);
            line-height: vw(17);
            font-weight: 500;
            overflow: hidden;
            color: black;
            cursor: pointer;
            .expand-icon-containers{
                background-color: map-get($colors, primary);
                position: absolute;
                right: 0;
                height: 100%;
                width: vw(38.07);
                display: flex;
                align-items: center;
                justify-content: center;
                border-bottom-right-radius: vw(5);
                border-top-right-radius: vw(5);
                .expand-icons{
                    font-size: vw(25);
                }
            }
        }
    }
}

.list-items-filter {
    list-style-type: none;
    margin: 0;
    padding: vw(10) 0 vw(20) 0;
    top: vw(40);
    right: 0px;
    width: vw(203.22);
    height: vw(185);
    overflow-y: scroll;
    background-color: white;
    position: absolute;
    box-shadow: 0px 0.3vw 0.6vw 0px rgba(0,0,0,0.2);
    z-index: 1;
    font-family: 'Lato-Bold';
    font-size: vw(14);
    line-height: vw(17);
    font-weight: 500;
    color: #3C3B3B;
    .list-item-filter {
        padding: vw(6) vw(19);
        color: #3C3B3B;
        font-family: 'Lato-Bold';
        font-size: vw(14);
        line-height: vw(17);
    }
    .list-item-filter:hover {
        background-color: #e5e5e5;
        color: #3C3B3B;
    }
}

@media (min-width: 600px) and (max-width: 1025px) {
    .dropdown-filter {
        position: relative;
        display: inline-block;
        background-color: #ffffff;
        width: vw-tab(220);
        height: vw-tab(33);
        border-radius: vw-tab(5);
        box-shadow: 0px 0.2vw 0.6vw rgba(0, 0, 0, 0.161);
        .dropdown-filter-button {
            position: relative;
            border-radius: vw-tab(5);
            display: flex;
            align-items: center;
            height: vw-tab(33);
            .dropdown-selected-items{
                margin: 0 0 0 vw-tab(19);
                flex: 1;
                position: relative;
                height: vw-tab(33);
                display: flex;
                flex-direction: row;
                white-space: nowrap;
                align-items: center;
                font-family: 'Lato-Medium';
                font-size: vw-tab(14);
                line-height: vw-tab(17);
                font-weight: 500;
                overflow: hidden;
                color: black;
                .expand-icon-containers{
                    background-color: map-get($colors, primary);
                    position: absolute;
                    right: 0;
                    top: 0;
                    height: vw-tab(33);
                    width: 20%;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    border-bottom-right-radius: vw-tab(5);
                    border-top-right-radius: vw-tab(5);
                    .expand-icons{
                        font-size: vw-tab(25);
                    }
                }
            }
        }    
        .list-items-filter {
            list-style-type: none;
            margin: 0;
            padding: vw-tab(10) 0 0 0;
            top: vw-tab(35);
            right: 0px;
            width: vw-tab(220);
            height: vw-tab(170);
            overflow-x: hidden;
            overflow-y: scroll;
            background-color: white;
            position: absolute;
            box-shadow: 0px 0.3vw 0.6vw 0px rgba(0,0,0,0.2);
            z-index: 1;
            font-weight: 500;
            color: #3C3B3B;
            .list-item-filter {
                padding: vw-tab(12) vw-tab(19);
                font-family: 'Lato-Medium';
                font-size: vw-tab(14);
                line-height: vw-tab(17);
                text-overflow: ellipsis;
                color: black;
                font-size: vw-tab(14);
            }
            .list-item-filter:hover {
                background-color: #e5e5e5;
                color: #000;
            }
        }
    }
}

@media (max-width: 599px) {
    .list-items-filter {
        list-style-type: none;
        display: flex;
        flex-direction: column;
        margin-bottom: vw-mobile(2);
        top: vw-mobile(2);
        padding: vw-mobile(10) 0 0 0;
        width: 100%;
        height: vw-mobile(170);
        overflow-x: hidden;
        overflow-y: scroll;
        background-color: white;
        border-radius: vw-mobile(5);
        position: relative;
        box-shadow: 0px vw-mobile(3) vw-mobile(6) rgba(0,0,0,0.2);
        font-weight: 500;
        color: #3C3B3B;
        .list-item-filter {
            padding: vw-mobile(12) vw-mobile(19);
            font-family: 'Lato-Medium';
            font-size: vw-mobile(14);
            line-height: vw-mobile(17);
            text-overflow: ellipsis;
            color: black;
            font-size: vw-mobile(14);
        }
        .list-item-filter:hover {
            background-color: #e5e5e5;
            color: #000;
        }
    }
    .dropdown-filter {
        position: relative;
        display: inline-block;
        background-color: #ffffff;
        width: 100%;
        height: vw-mobile(33);
        border-radius: vw-mobile(5);
        box-shadow: 0px vw-mobile(3) vw-mobile(6) rgba(0,0,0,0.2);
        .dropdown-filter-button {
            position: relative;
            border-radius: vw-mobile(5);
            display: flex;
            align-items: center;
            height: vw-mobile(33);
            .dropdown-selected-items{
                margin: 0 0 0 vw-mobile(19);
                height: vw-mobile(33);
                position: relative;
                flex: 1;
                display: flex;
                flex-direction: row;
                white-space: nowrap;
                align-items: center;
                font-family: 'Lato-Medium';
                font-size: vw-mobile(14);
                line-height: vw-mobile(17);
                font-weight: 500;
                overflow: hidden;
                color: black;
                .expand-icon-containers{
                    background-color: map-get($colors, primary);
                    position: absolute;
                    right: 0;
                    top: 0;
                    height: vw-mobile(33);
                    width: 20%;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    border-bottom-right-radius: vw-mobile(5);
                    border-top-right-radius: vw-mobile(5);
                    .expand-icons{
                        font-size: vw-mobile(25);
                    }
                }
            }
        }            
    }
}