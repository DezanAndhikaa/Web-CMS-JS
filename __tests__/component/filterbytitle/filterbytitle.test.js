import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import FilterByTitle from 'components/FilterByTitle/DropdownFilter';

Enzyme.configure({adapter:new Adapter()});
it("render filter by title correctly", () => {
    const wrapper = shallow (
        <FilterByTitle />
    )
    expect(wrapper).toMatchSnapshot();
})