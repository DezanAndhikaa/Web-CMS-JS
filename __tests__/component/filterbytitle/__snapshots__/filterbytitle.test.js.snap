exports[`render filter by title correctly 1`] = `
<div
  className="dropdown-filter"
>
  <div
    className="dropdown-filter-button"
    onClick={[Function]}
  >
    <div
      className="dropdown-selected-items"
    >
      <div
        className="expand-icon-containers"
      >
        <ExpandMoreIcon
          className="expand-icons"
        />
      </div>
    </div>
  </div>
</div>
`;